presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedSet.java

```diff
@@ -130,9 +134,29 @@ public void testBigintSimpleTypedSet()
         }
     }
 
+    @Test
+    public void testMemoryExceeded()
```

Comment:
per our discussion on the perils of statics, if max memory for TypedSet were not a static var, this would be much simpler to test
(though we might not understand why we are seeing such high memory use on a large sql array be parsed)

i'll keep digging

but changing to

array_distinct(ARRAY['some relally long string') did not trigger the failure. it never reaches TypedSet code (I thnk because it short-circuits that there is only 1 item, and optimizes that way)

working on

array_distinct(ARRAY['one big string',  'one other big string'])

to see how that flies
i know it fails if i pas in tons of small elements (ints)
but needs like an 8g heap (which is odd) and takes ~2 m